### DISCUSSION TITLE
Mutation update how pass null value

### DISCUSSION THREAD

**Discussion by juliocebrito** [graphql-python/graphene-django#1184] [Q&A] [answered]
**model**
`
date = models.DateField(blank=True, null=True)
`
**schema**
`
date = graphene.types.datetime.Date()
`

`
mutation ($var1: Date){
  updateModel(
    id: 1,
    date: $var1
  ) {
    model {
      id
      date
    }
  }
}
`

`
{"var1": null}
`

`
{
  "errors": [
    {
      "message": "mutate() missing 1 required positional argument: 'date'",
      "locations": [
        {
          "line": 2,
          "column": 3
        }
      ]
    }
  ],
  "data": {
    "updateModel": null
  }
}
`

**Comment by spockNinja** [comment, marked as answer]
@juliocebrito 

I haven't tried this myself recently on the latest graphene/graphene-django versions. But based on [these docs for graphene](http://docs.graphene-python.org/en/latest/types/mutations/#inputfields-and-inputobjecttypes), I believe you should be able to do the following:

```python

class UpdateModel(graphene.Mutation):
    class Arguments:
        id = graphene.ID()
        date = graphene.Date(required=False)

    @staticmethod
    def mutate(root, info, id, date=None):
        # handle empty date
```

Let me know if that answers your question and works for you.

**Comment by Eraldo** [comment]
Sadly this does not work for me using 2.0.1. :|

I keep getting `None` when passing `null` in my client. (vs `undefined`)

**Comment by dopeboy** [comment]
@Eraldo can you share a relevant code snippet?

**Comment by dopeboy** [comment]
Closing this down due to inactivity.

**Comment by isakal** [comment]
> @juliocebrito
> 
> I haven't tried this myself recently on the latest graphene/graphene-django versions. But based on [these docs for graphene](http://docs.graphene-python.org/en/latest/types/mutations/#inputfields-and-inputobjecttypes), I believe you should be able to do the following:
> 
> ```python
> class UpdateModel(graphene.Mutation):
>     class Arguments:
>         id = graphene.ID()
>         date = graphene.Date(required=False)
> 
>     @staticmethod
>     def mutate(root, info, id, date=None):
>         # handle empty date
> ```
> 
> Let me know if that answers your question and works for you.

This worked for me 
Thanks!
